src/main/java/seedu/programmer/ui/CommandBox.java

```diff
@@ -59,15 +60,21 @@ private void handleCommandEntered() {
      */
     @FXML
     private void handleKeyPressed(KeyEvent event) {
-        if (event.getCode() == KeyCode.UP) {
-            commandTextField.setText(commandHistory.getPrevCommand());
-        } else if (event.getCode() == KeyCode.DOWN) {
-            commandTextField.setText(commandHistory.getNextCommand());
-        } else {
+        try {
+            if (event.getCode() == KeyCode.UP) {
+                commandTextField.setText(commandHistory.getPrevCommand());
+            } else if (event.getCode() == KeyCode.DOWN) {
+                commandTextField.setText(commandHistory.getNextCommand());
+            } else {
+                return;
+            }
+            setStyleToDefault();
+            commandTextField.end();
+            event.consume(); // Consume Event
+        } catch (CommandHistoryException e) {
+            commandTextField.end();
```

Comment:
Not sure if this is the best implementation to solve this from a code quality perspective, because compared to the previous implementation now we can't really tell what went "wrong" if a CommandHistory Exception were to occur. It also seems like we are using the catching of an exception to carry out the expected functionality.

Reply:
We could add a log here, but my gut feeling is that a new exception here isn't really necessary. Maybe I can try an alternative method and make a PR? I'm thinking it is possible to do this with if-else conditional statements.

Reply:
Thanks man! Very true, I am apprehensive of using an exception here as a 'control flow'.
@AllardQuek yes sure, you can pull this branch to your local and give it a try and just push back and it will update in this PR. I will consider what I can possibly do and discuss with you offline to see what could work best. Thanks a lot!

Reply:
Ok sure! Is it ok if I push to the same branch? I created a branch from this branch to work on just in case. I'll make a separate PR with that first and mention this PR, in case you want to keep a record of your implementation. Thanks :D